inst/sympref.m

```diff
@@ -183,140 +183,162 @@
     return
   end
 
-
-  switch lower(cmd)
-    case 'defaults'
-      settings = [];
-      settings.ipc = 'default';
-      settings.whichpython = '';
-      sympref ('display', 'default')
-      sympref ('digits', 'default')
-      sympref ('quiet', 'default')
-
-    case 'version'
-      assert (nargin == 1)
-      varargout{1} = '2.5.0-dev';
-
-    case 'display'
-      if (nargin == 1)
-        varargout{1} = settings.display;
+  if (isstruct(cmd))
+    % Means we need to copy it
+    s = sympref();
+    names = fieldnames (s);
+
+    if (sum(isfield(cmd, names)) == numfields(s))
+      if (typeinfo(cmd.ipc) == typeinfo(s.ipc) && ...
```

Comment:
Instead of these two `if`s, how about:
````
assert (isempty (xor (fieldnames (s),   {'ipc', 'whichpython', 'display', 'digits', 'quiet'})), ...
        'sympref: Type of fields of the structure is not correct')
````